### ISSUE TITLE
License conflicts

### ISSUE THREAD

**Issue by timmwille** [databrickslabs/dolly#37]
I've shared this as Open Source news but then had a look at your data sets license. This is not open source, so I see many conflicts here. NC is not a Non-Profit license (would be nice, I know) it actually forbids any commercial activity. Even getting payed as a researcher when using it. 

So I consider this an epic issue to solve → furthermore, it blocks contributions as they would fall under the same conditions and would completely lead the whole effort to crowdsource ad adsurdum. (also you as a team can then be not commercially involved at any point) → so I admire your efforts to create something beyond ChatGPT and would like to continue spreading the message, if you could carefully reconsider the license approach to be truly open source.

All the best

**Comment by timmwille** [comment]
might be linked to discuss with them?
https://crfm.stanford.edu/2023/03/13/alpaca.html

or LLaMA directly? 

this will run into lots of conflicts in the future

**Comment by DutKulang** [comment]
The licence might bring up issues later in the future better corrected early, thanks @timmwille for bring the matter to the spot light

**Comment by timmwille** [comment]
If anyone can explain how deep this goes, I would appreciate, this is what I found so far:
https://github.com/tatsu-lab/stanford_alpaca#data-generation-process

And a lawsuit with Meta aka Facebook mother firm is as bad as using ChatGPT with Microsoft spinning the wheel for the OS Community (I might be wrong here, but it has a bitter taste for freedom IMHO)
https://github.com/facebookresearch/llama

**Comment by srowen** [comment]
I'm not quite sure what the question is. There is no data set released in this repo. 
The source code is Apache licensed.

The model you train from this code is not based on LLaMa. It is based on a data set that is ultimately from OpenAI output, which carries some restrictions on use. I'm sure eventually everyone will switch to open data alternatives.

Does that help?

**Comment by timmwille** [comment]
> I'm not quite sure what the question is. There is no data set released in this repo. The source code is Apache licensed.
> 
> The model you train from this code is not based on LLaMa. It is based on a data set that is ultimately from OpenAI output, which carries some restrictions on use. I'm sure eventually everyone will switch to open data alternatives.
> 
> Does that help?

yes it helps, I'm not sure this would even hold stand against any court if the above is true. I'm not a lawyer so I just wonder if it is part of the roadmap to leave that wonky trail of NC. Do you have a direct link to the closed data from LLaMA? I can check but yes I agree, the precise question would be: what alternatives are planned or what are the requirements to fit Dolly?

**Comment by srowen** [comment]
I'm not sure what you're referring to. Would what stand up?
As I say, this is not related to LLaMa.
Also as I say, sure, I'm sure people will rapidly find an alternative to the OpenAI-sourced data and retrain. It doesn't really require that data.

**Comment by timmwille** [comment]
I think this is still not solved as the readme states: 

`Dolly is intended exclusively for research purposes and is not licensed for commercial use.`

don't get me wrong, I share your non-profit orientation though it has in a way conflicts with the license implementation that is not directed towards "non-profit" but copyleft at least so close, and the "non-commercial" passus conflicts there in all definitions of freedom within the OSI standard. 

I think the intention is noble but I think you can not solve it via NC clause as this also considers contributions on unstable ground.

---

**Comment by srowen** [comment]
To be clear, this is driven entirely by the training data set's nature, not anything else here like the code or base model. And you can imagine that soon this won't even depend on this particular encumbered data set. There is no copy-left license here. 

In plain English, are you just saying you want something more like the ALv2 license? sure, sit tight.
